engine/src/flutter/runtime/dart_isolate.cc

```diff
@@ -153,11 +153,15 @@ std::weak_ptr<DartIsolate> DartIsolate::CreateRunningRootIsolate(
     return {};
   }
 
-  if (settings.root_isolate_create_callback) {
-    // Isolate callbacks always occur in isolate scope and before user code has
-    // had a chance to run.
+  {
     tonic::DartState::Scope scope(isolate.get());
-    settings.root_isolate_create_callback(*isolate.get());
+    Dart_SetCurrentThreadOwnsIsolate();
+
+    if (settings.root_isolate_create_callback) {
+      // Isolate callbacks always occur in isolate scope and before user code
+      // has had a chance to run.
+      settings.root_isolate_create_callback(*isolate.get());
+    }
   }
 
   if (root_isolate_create_callback) {
```

Comment:
This patch lgtm but I got totally confused by the two "root isolate create callbacks". One in the settings, and another passed as an argument. Why are there two? Why does only one run in isolate scope? Why can't one wrap the other?

Reply:
Not sure. I've only touched this code a couple of times 😅